src/index.js

```diff
@@ -31,15 +33,27 @@ function installFiles(sourceDir, done) {
   // The path to the package running the 'install' or 'postinstall' script.
   var fileInstallingPackagePath = hostPackageDir(scriptPath);
 
-  // The path to the package into which we should install the files.
-  var destinationDir = fileInstallingPackagePath && hostPackageDir(fileInstallingPackagePath);
-  if (!destinationDir) {
+  var source, target;
+  switch (npmv.majorVersion()) {
+    case '3':
```

Comment:
I just tested and `process.env.npm_config_user_agent` is set/of the same format in v2, so all clear to invert the cases.